hdk/virtual_machine/syntax.py

```diff
@@ -0,0 +1,75 @@
+"""Defines dataclasses that represent different types of vm instructions."""
+from dataclasses import dataclass
+from typing import ClassVar, TypeAlias
+
+
+@dataclass(frozen=True)
+class StackInstruction:
+    """A data class representing a stack instruction in a virtual machine.
+
+    Attributes:
+        command: The operation to be performed on the stack.
+    """
+
+    _ALLOWED_COMMAND: ClassVar[set[str]] = {
+        "add",
+        "sub",
+        "neg",
+        "eq",
+        "gt",
+        "lt",
+        "and",
+        "or",
+        "not",
+    }
+
+    command: str
+
+    def __post_init__(self):
+        cls = self.__class__
+        if self.command not in cls._ALLOWED_COMMAND:
+            raise ValueError(f"Wrong command {self.command!r} for operation on stack.")
+
+
+@dataclass(frozen=True)
+class MemoryAccessInstruction:
```

Comment:
Rename this class to MemoryTransferCommand and change the docstring to start with "Represents...".